source/_integrations/modbus.markdown

```diff
@@ -198,6 +198,19 @@ Description:
 | value     | (write_register) A single value or an array of 16-bit values. Single value will call modbus function code 0x06. Array will call modbus function code 0x10. Values might need reverse ordering. E.g., to set 0x0004 you might need to set `[4,0]`, this depend on the byte order of your CPU |
 | state     | (write_coil) A single boolean or an array of booleans. Single boolean will call modbus function code 0x05. Array will call modbus function code 0x0F |
 
+### Example: writing a float32 type register
+
+To write a float32 datatype register use network format like `10.0` == `0x41200000` (network order float hexadecimal). Call the modbus.write_register:
+
+```yaml
+service: modbus.write_register
+data:
+  address: <target register address>
+  unit: <target slave address>
+  hub: <hub name>
```

Comment:
```suggestion
```
When defining examples with placeholders only show required fields

Reply:
But unit is required, and hub should be (it about to be corrected in an upcoming PR).
```
SERVICE_WRITE_COIL_SCHEMA = vol.Schema(
    {
        vol.Optional(ATTR_HUB, default=DEFAULT_HUB): cv.string,
        vol.Required(ATTR_UNIT): cv.positive_int,
        vol.Required(ATTR_ADDRESS): cv.positive_int,
        vol.Required(ATTR_STATE): vol.Any(
            cv.boolean, vol.All(cv.ensure_list, [cv.boolean])
        ),
    }
)
```

Reply:
If that is required, the documentation is wrong.

> unit: Slave address (1-255, defaults to 0)

Reply:
you got me ! ! I will update the documentation p.

Reply:
please see https://github.com/home-assistant/home-assistant.io/pull/19313